static/app/components/forms/compactSelect.tsx

```diff
@@ -25,7 +28,11 @@ interface TriggerRenderingProps {
   ref: React.RefObject<HTMLButtonElement>;
 }
 
-interface Props extends ControlProps, Partial<OverlayProps>, Partial<AriaPositionProps> {
+interface Props<OptionType>
+  extends Omit<ControlProps<OptionType>, 'choices'>,
+    Partial<OverlayProps>,
+    Partial<AriaPositionProps> {
+  options: Array<OptionType & {options?: OptionType[]}>;
```

Comment:
adding support for the `OptionType` generic, same as in [`SelectControl`](https://github.com/getsentry/sentry/blob/vl/compact-select-sections/static/app/components/forms/selectControl.tsx#L64)